lib/proxy/peer/client.go

```diff
@@ -233,9 +237,25 @@ func (c *grpcClientConn) Shutdown(ctx context.Context) {
 
 // Close implements [internal.ClientConn].
 func (c *grpcClientConn) Close() error {
+	c.pingCancel()
 	return c.cc.Close()
 }
 
+// Ping implements [internal.ClientConn].
+func (c *grpcClientConn) Ping(ctx context.Context) error {
+	release := c.maybeAcquire()
+	if release == nil {
+		return trace.ConnectionProblem(nil, "error starting stream: connection is shutting down")
+	}
+	defer release()
+
+	_, err := clientapi.NewProxyServiceClient(c.cc).Ping(ctx, new(clientapi.ProxyServicePingRequest))
+	if trace.IsNotImplemented(err) {
+		err = nil
```

Comment:
i was thinking about the handling of this, but I think you are right that getting not implemented from the server indicates a successful ping